sql/parser/sql.y

```diff
@@ -887,19 +890,25 @@ opt_boolean_or_string:
 
 // Timezone values can be:
 // - a string such as 'pst8pdt'
-// - an identifier such as "pst8pdt"
-// - an integer or floating point number
 // - a time interval per SQL99
-// name gives reduce/reduce errors against const_interval and LOCAL, so use
-// IDENT (meaning we reject anything that is a key word).
 zone_value:
-  SCONST { unimplemented() }
-| IDENT { unimplemented() }
-| const_interval SCONST opt_interval { unimplemented() }
-| const_interval '(' ICONST ')' SCONST { unimplemented() }
-| numeric_only { unimplemented() }
```

Comment:
We certainly allow casting an `int` to an `interval`, but note that this is somewhat different. The number here is in units of hours. I'm not sure why there would be more confusion to `SET TIME ZONE -7` vs `SET TIME ZONE INTERVAL '-7h'`.